consensus/types/src/data_column_sidecar.rs

```diff
@@ -32,6 +34,51 @@ pub struct DataColumnIdentifier {
     pub index: ColumnIndex,
 }
 
+#[derive(Serialize, Deserialize, Encode, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DataColumnsByRootIdentifier {
+    pub block_root: Hash256,
+    pub indices: RuntimeVariableList<ColumnIndex>,
+}
+
+impl RuntimeVariableList<DataColumnsByRootIdentifier> {
+    pub fn from_ssz_bytes_with_nested(
+        bytes: &[u8],
+        max_len: usize,
+        num_columns: usize,
+    ) -> Result<Self, DecodeError> {
+        if bytes.is_empty() {
+            return Ok(RuntimeVariableList::empty(max_len));
+        }
+
+        let vec = ssz::decode_list_of_variable_length_items::<Vec<u8>, Vec<Vec<u8>>>(
+            bytes,
+            Some(max_len),
+        )?
+        .into_iter()
+        .map(|bytes| {
+            // Split manually: first 32 bytes = block_root and next 4 bytes = vector tag
```

Comment:
I think it would be a bit cleaner to use `SszDecoderBuilder` here to decode, and call into a separate `DataColumnByRootIdentifier` decode function

Reply:
Something like this:

```rust
let mut builder = ssz::SszDecoderBuilder::new(&bytes);
builder.register_type::<Hash256>()?;
builder.register_anonymous_variable_length_item()?;

let mut decoder = builder.build()?;
let block_root = decoder.decode_next()?;
let indices = decoder.decode_next_with(|bytes| DataColumnsByRootIdentifier::from_ssz_bytes(bytes, num_columns))?;
Ok(DataColumnsByRootIdentifier {
    block_root,
    indices,
})
```